activerecord/lib/active_record/associations/collection_association.rb

```diff
@@ -642,6 +644,14 @@ def first_nth_or_last(type, *args)
             set_inverse_instance record if record.is_a? ActiveRecord::Base
           end
         end
+
+        def associate_all_records(target)
+          if owner.new_record? && target.all? {|record| record.new_record? }
```

Comment:
```
car = Car.new(seats: [Seat.new, Seat.find(1)])
car.seats.first.car == car
# => true
```

The `car` has not yet been saved or associated with any of the seats, so how come the above comparison returns `true`?